libbindgen/tests/headers/disable-namespacing.hpp

```diff
@@ -0,0 +1,9 @@
+// bindgen-flags: --disable-name-namespacing
+
+namespace foo {
+namespace bar {
+
+typedef int Baz;
```

Comment:
What happens when we have a `foo::Baz` as well? Do we generate incorrect code? It would be nice to fail loudly or re-use the overloading numbers, or something. I don't care much *which* approach we choose, but we should explicitly choose one, document its behavior, and properly handle it as we've decided in the code.